jwst/cube_build/tests/test_configuration.py

```diff
@@ -467,3 +539,41 @@ def test_calspec3_config_nirspec_multi(tmp_cwd, nirspec_medium_coverage):
 
     assert cube_pars['1']['par1'] == ['g140m', 'g235m']
     assert cube_pars['1']['par2'] == ['f100lp', 'f170lp']
+
```

Comment:
These look like good tests for the check_offset_file function, but none of the code that actually applies the offsets is covered.  It looks like there is some synthetic data in some of the other tests that can be run through cube_build successfully - can you write a test or two that applies some example offsets to the synthetic data?

Reply:
Would it be sufficient to make a regression tests to test applying the offsets ? I am trying to set up a unit test but we do not call the full cube building in the unit tests - we call smaller subsets. The cube_building is done in ifu_cube.py.  If we call ifu_cube.setup_ifucube_wcs - then we could test how the cube coordinate system is shifted. But there is still quite a lot in that routine and it not all the easy form a unit test because it  requires so of the infrastructure to be setup - WCS, WCSINFO, models. Just checking what the real goal is and if a regression test would be sufficient.  I could rewrite setup_ifube_wcs to do less set up  so it is easier to write unit test for it- but not sure that is a good idea either.

Reply:
Two goals...

One, most of the code you've added is not currently covered by either regression or unit tests, which makes me nervous.  I think we should add one or the other.

Two, unit test coverage for this module looks pretty minimal in general.  It would be nice to start working on getting that coverage up, so if it's possible to add some unit tests here, it would help with maintainability and robustness in this part of the code base.

So, if possible, it would be nice to add unit tests to cover these changes.  If it is too difficult to manage within the scope of this PR, then we should add a regression test.

It looks to me like there are some tests in test_cube_build_step.py that run the full step, which is why I thought you might be able to build on those and create a test that passes in an offset file.  But I haven't tried, so it may be more difficult than it looks!

Reply:
I finally have it. I just had to set up more default values to the code so it would run all the way to test ifu_cube.setup_ifucube_wcs